sklearn/utils/estimator_checks.py

```diff
@@ -1505,7 +1505,8 @@ def _check_sample_weight_equivalence(name, estimator_orig, sparse_container):
 
     rng = np.random.RandomState(42)
     n_samples = 15
-    X = rng.rand(n_samples, n_samples * 2)
+    X = rng.rand(n_samples, n_samples * 2) * 5
+    X = _enforce_estimator_tags_X(estimator_orig, X)
```

Comment:
Note: I have pushed this change in the PR to make sure that this check is meaningful enough for estimators that accept categorical features as inputs.